### PR TITLE
Accessibility: removing accessibility quirks

### PR DESCRIPTION
**Pull request by M-Lipin**
contributes to #485

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs [hunk 1]
```diff
@@ -71,17 +71,23 @@ internal abstract class ButtonBaseAdapter {
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         protected bool IsHighContrastHighlighted() {
-            return AccessibilityImprovements.Level1 && IsHighContrastHighlightedInternal();
+            // Initially calling IsHighContrastHighlightedInternal was dependendent on 
+            // the Accessibility Improvement 1 which were opted-in/out in configuration.
+            return IsHighContrastHighlightedInternal();
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs]: Looks like these can all be the same single function now.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs]: And the same applies to the following 2 functions

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs [hunk 2]
```diff
@@ -1525,15 +1523,7 @@ private string[] GetStringsForAutoComplete(IList collection)
         /// <devdoc>
         /// </devdoc>
         protected override AccessibleObject CreateAccessibilityInstance() {
-            if (AccessibilityImprovements.Level3) {
-                return new ComboBoxUiaProvider(this);
-            }
-            else if (AccessibilityImprovements.Level1) {
-                return new ComboBoxExAccessibleObject(this);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs]: Can we remove the "old" accessible object classes?

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 3]
```diff
@@ -8623,9 +8622,7 @@ private static void InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
         protected void InvokeGotFocus(Control toInvoke, EventArgs e) {
             if (toInvoke != null) {
                 toInvoke.OnGotFocus(e);
-                if (!AccessibilityImprovements.UseLegacyToolTipDisplay) {
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: I'm not sure about this one, this feature might be annoying to the users. Had you discussed this with Merrie?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs [hunk 4]
```diff
@@ -262,73 +262,58 @@ internal override int[] RuntimeId
 
             internal override bool IsIAccessibleExSupported()
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: Had you reviewed if this property is still needed or can be removed now? Is it still `false` for some classes?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs [hunk 5]
```diff
@@ -262,73 +262,58 @@ internal override int[] RuntimeId
...
-                            return true;
-                        case NativeMethods.UIA_ControlTypePropertyId:
-                            return NativeMethods.UIA_TableControlTypeId;
-                        case NativeMethods.UIA_ItemStatusPropertyId:
-                            // Whether the owner DataGridView can be sorted by some column.
-                            // If so, provide not-sorted/sorted-by item status.
-                            bool canSort = false;
-                            for (int i = 0; i < owner.Columns.Count; i++)
+                    case NativeMethods.UIA_NamePropertyId:
+                        return this.Name;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: FYI - `this.` can be removed in core. However it's nice to keep the change focused on the feature  - quirks removal, and do clean up later. Lessens chance of missing a bug :)

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs [hunk 6]
```diff
@@ -4770,20 +4764,6 @@ public override string DefaultAction
...
             }
 
-            /// <include file='doc\DataGridViewCell.uex' path='docs/doc[@for="DataGridViewCellAccessibleObject.Help"]/*' />
-            public override string Help
-            {
-                get
-                {
-                    if (AccessibilityImprovements.Level2)
-                    {
-                        return null;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: Why did you remove this completely? Is `null` the default?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxEditingControl.cs [hunk 7]
```diff
@@ -31,16 +31,7 @@ public DataGridViewComboBoxEditingControl() : base()
         /// <include file='doc\DataGridViewComboBoxEditingControl.uex' path='docs/doc[@for="DataGridViewComboBoxEditingControl.CreateAccessibilityInstance"]/*' />
         protected override AccessibleObject CreateAccessibilityInstance()
         {
-            if (AccessibilityImprovements.Level3)
-            {
-                return new DataGridViewComboBoxEditingControlAccessibleObject(this);
-            }
-            else if (AccessibilityImprovements.Level2)
-            {
-                return new DataGridViewEditingControlAccessibleObject(this);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxEditingControl.cs]: Can this class be removed now?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxCell.cs [hunk 8]
```diff
@@ -56,12 +56,7 @@ public DataGridViewTextBoxCell()
         /// </returns>
         protected override AccessibleObject CreateAccessibilityInstance()
         {
-            if (AccessibilityImprovements.Level1)
-            {
-                return new DataGridViewTextBoxCellAccessibleObject(this);
-            }
-
-            return base.CreateAccessibilityInstance();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxCell.cs]: is the "old" class still needed?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.cs [hunk 9]
```diff
@@ -336,9 +324,6 @@ private static HorizontalAlignment TranslateAlignment(DataGridViewContentAlignme
     /// <summary>
     /// Defines the DataGridView TextBox EditingControl accessible object.
     /// </summary>
-    /// <remarks>
-    /// This accessible object is only available in AccessibilityImprovements of Level 3.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.cs]: nice job cleaning up comments!

**Comment by zsd4yr** [comment]
I'll take another look once the current comments have been addressed

File: src/System.Windows.Forms/src/System/Windows/Forms/LinkLabel.cs [hunk 10]
```diff
@@ -2660,22 +2656,13 @@ internal class LinkAccessibleObject : AccessibleObject {
...
-                        int charStart = LinkLabel.ConvertToCharIndex(link.Start, text);
-                        int charEnd = LinkLabel.ConvertToCharIndex(link.Start + link.Length, text);
-                        name = text.Substring(charStart, charEnd - charStart);
-                        if (AccessibilityImprovements.Level1 && link.Owner.UseMnemonic) {
-                            // return the same value as the tooltip shows.
-                            name = WindowsFormsUtils.TextWithoutMnemonics(name);
-                        }
+
+                    // return the full name of the link label
+                    // assometimes the link name in isolation
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/LinkLabel.cs]:
```suggestion
                    // as sometimes the link name in isolation
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs (hunk 4 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: It seems that we cannot remove this at this moment. This is false by default and not overridden to be true in several classes. So if we remove it and make true by default, the behavior can be changed (this is risky for this PR to me), but we can file additional task to thoroughly review all the places where this is false and fix this in the context of another work item.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs (hunk 6 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: I removed this because GetPropertyValue returns this.Help or empty string when getting UIA_HelpTextPropertyId property from DatgaGridViewCellAccessibleObject.

Before the change Help property always returns null for Level >= Level2 and GetPropertyValue(HelpTextPropertyId) always returns empty string for Level >= Level2;

In case we are removing this property Help property always returns null (as AccessibleObject.Help returns null if systemIAccessible == null).
GetPropertyValue(HelpTextPropertyId) always returns empty string as base.Help == null.

Maybe I'm missing something, but it seems to me that the change is of and the help is null by default for DataGridViewCell. Please correct me if I'm wrong.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxEditingControl.cs (hunk 7 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewComboBoxEditingControl.cs]: Yes, thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.cs (hunk 9 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxEditingControl.cs]: thanks :)

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs (hunk 5 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: Thank you for pointing on this, Tanya!

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs (hunk 1 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonInternal/ButtonBaseAdapter.cs]: Thanks, fixed this in fcea72c6b9d9a39ea045a2a2ca3246078a9ac1f6.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs (hunk 2 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs]: Removed this is fcea72c6b9d9a39ea045a2a2ca3246078a9ac1f6. Thanks!

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: Filed this: https://github.com/dotnet/winforms/issues/802.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxCell.cs (hunk 8 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxCell.cs]: Do you mead DataGridViewCellAccessibleObject? This is still used.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTextBoxCell.cs]: sounds good

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs (hunk 6 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: thank you for the details

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs (hunk 4 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: sounds good

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: thanks!

**Comment by Tanya-Solyanik** [comment]
Hi @M-Lipin ,Thank you for making changes and adding details! Looks good, let Zack to review his requested change, and then merge!

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs [hunk 11]
```diff
@@ -5029,7 +5029,7 @@ internal override bool SupportsUiaProviders
         {
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs]: This can be simplified 

```cs
        internal override bool SupportsUiaProviders	
            => true;
```

this applies for other instances of this override to true

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs (hunk 4 above)
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: Can also be simplified

```cs
        internal override bool IsIAccessibleExSupported()	
            => true;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs [hunk 12]
```diff
@@ -7060,18 +7060,13 @@ internal override bool SupportsUiaProviders
             {
                 get
                 {
-                    return AccessibilityImprovements.Level3;
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs]: can be simplified

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewButtonCell.cs [hunk 13]
```diff
@@ -1137,12 +1137,7 @@ public override int GetChildCount()
 
             internal override bool IsIAccessibleExSupported()
             {
-                if (AccessibilityImprovements.Level2)
-                {
-                    return true;
-                }
-
-                return base.IsIAccessibleExSupported();
+                return true;
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewButtonCell.cs]: can be simplified

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs (hunk 11 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs]: Thank you for this point! Fixed.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs (hunk 12 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs]: Thanks!

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs (hunk 4 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewAccessibleObject.cs]: thank you, fixed

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewButtonCell.cs (hunk 13 above)
Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewButtonCell.cs]: fixed, thanks

**Comment by Tanya-Solyanik** [comment]
@zsd4yr  - could you please take another look?

**Comment by zsd4yr** [comment]
Sorry thought i had already approved
